Handle the options passed to llvm opt and llvm llc

chentong319 opened the pull request:
Issue #2950 discovered the issue in passing options llvm. There are several problems:
1. The string of the options has to be split into vector of strings for the command format.
2. Conflict in options. So far, the optimization level is the only example.  The level specified with  particular step should have higher priority than that for the whole compiler. Thus, I can avoid multiple "-O#" error.
3. modify the comments for mllvm, xopt, and xllc to make it easy for user to pass multiple values to these options 

I tested with some examples.

tungld commented:
> Options of -mllvm is for both llvm opt and llc, but their options are not the same. New option, -mllvmopt and -mllvmllc, are introduced to pass specific option to opt or llc respectively.

Look like `-mllvmopt` and `-mllvmllc` are similar to two existing options `-Xopt` and `-Xllc`.

chentong319 commented:
> > Options of -mllvm is for both llvm opt and llc, but their options are not the same. New option, -mllvmopt and -mllvmllc, are introduced to pass specific option to opt or llc respectively.
> 
> Look like `-mllvmopt` and `-mllvmllc` are similar to two existing options `-Xopt` and `-Xllc`.

That's true. The only difference is that each value of opt and xllc has the same limitation of mllvm: only one option is allowed. I removed the new llvmopt and llvmllc, and modified the option explanation for opt and xllc.

tungld reviewed (approved):
LGTM.
